packages/plugin-vue-jsx/src/index.ts

```diff
@@ -21,10 +21,13 @@ export function pluginVueJsx(options: PluginVueJsxOptions = {}): RsbuildPlugin {
           CHAIN_ID,
           modifier: (babelOptions) => {
             babelOptions.plugins ??= [];
-            babelOptions.plugins.push([
-              require.resolve('@vue/babel-plugin-jsx'),
-              options.vueJsxOptions || {},
-            ]);
+            babelOptions.plugins.push(
+              [
+                require.resolve('@vue/babel-plugin-jsx'),
+                options.vueJsxOptions || {},
+              ],
+              [require.resolve('babel-plugin-vue-jsx-hmr')],
```

Comment:
In some cases we do not need to add this plugin. For example, when `dev.hmr` is disabled, or when target is not `web`.

You can use the `isUsingHMR` helper from `@rsbuild/shared` to check whether HMR is enabled.

Reply:
Yes, that's right. The hmr feature is only available when dev.hmr is enabled.

Reply:
[1f85277](https://github.com/web-infra-dev/rsbuild/pull/2018/commits/1f852778b2b01c901664fb322cf68bca3987d070)
[28c1e7a](https://github.com/web-infra-dev/rsbuild/pull/2018/commits/28c1e7a2b1062cfcf80a69d76c76992202ecce50)